packages/compass-schema/src/components/field.tsx

```diff
@@ -220,6 +214,12 @@ function Field({
   const fieldAccordionButtonId = `${JSON.stringify(path)}.${name}-button`;
   const fieldListRegionId = `${JSON.stringify(path)}.${name}-fields-region`;
 
+  // Convert the string array of paths
+  // to a single string with dots between field names.
+  // Note: This will cause collisions when
+  // there are fields with dots in them.
+  const fieldName = path.join('.');
```

Comment:
Which is a bug, right? Do you happen to know if we have a ticket?

Reply:
I'll try to find a reference and will open a ticket if I can't, thanks for calling this out

Reply:
I've written various functions that just do (simplified)

```javascript
path.map((part) => typeof part === 'string' ? `["${part}"]` : `[${part}]`).join('');
```

to try and avoid this. Then decide if you want to escape possible " chars in path. Maybe we need something "standard".

Reply:
Right, but that doesn’t help here because we are literally doing JS property accesses here, i.e. these strings are not just otherwise opaque identifiers for the fields

Reply:
yeah this case indeed needs more. I have functions for looking things up in an object by arrays of path bits too, though :)

Reply:
Added a TODO with a ticket number in https://github.com/mongodb-js/compass/pull/5244/commits/6b3cbf9cfd1a85ecc36f21304c215cc8d647370e